time-it-macros/src/lib.rs

```diff
@@ -36,11 +27,28 @@ pub fn time_it(to_time: TokenStream) -> TokenStream {
         let #start_ident = std::time::Instant::now();
         #to_time
         let duration = #start_ident.elapsed();
-        time_it::action(#tag_opt, duration);
+        time_it::run(#tag_opt, duration);
     }
     .into()
 }
 
+#[proc_macro_attribute]
+pub fn main(_: TokenStream, item: TokenStream) -> TokenStream {
+    let ast: ItemFn = parse(item).unwrap();
+    let fn_name = &ast.sig.ident;
+    let block = &ast.block;
+
+    let result = quote! {
+        fn #fn_name() {
+            #block
+            // Await the subscriber to finish processing
+            time_it::wait_for_finish();
+        }
+    };
```

Comment:
this will cause problem when there is a main which returns something.
1. the function signature of main has been changed since this new main doesn't have a return type
   1. i think you can just put the whole `ast.sig` here instead of just using the name
2. even if the return type is added the wait for finish will be after the return statement so it will not run. If they choose to return by just putting a floating expression then it will lead to a compilation error since you can't have any statements after the floating expression. I see two possible ways around this:
   1. Rename their main to something else which is unlikely to have name conflicts and then define your own main with the same signature which calls their main and then `wait_for_finish` and then returns the value from main. This is what happens in `#[time_fn]`
   2. Create a variable at the start of their main which impls drop and then the drop calls `wait_for_finish` this is guaranteed to be called regardless of where their function returns. I think I am going to change `#[time_fn]` to do something more like this so we don't have to define extra functions
<img width="652" alt="Screen Shot 2023-05-30 at 7 34 37 PM" src="https://github.com/tStreichenberger/time-it/assets/108824628/8fa8a043-1f58-4153-987f-55f56080ad2d">

Reply:
Looking

Reply:
Me when i create a macro to mess up the user's code